### DISCUSSION TITLE
Clarification on Error: Multiple maps with the same ID

### DISCUSSION THREAD

**Discussion by johnnyboustany** [uber/react-map-gl#2420] [Q&A]
### **Description:**
I encountered an error related to multiple maps having the same ID. The error is thrown in the onMapMount function in the MapProvider. I have a stable ID set for the map and everything is working as expected. However, I get the JS error "Multiple maps with the same ID" in the console. I am using reuseMaps to reduce the number of map instances being used.

### **Steps to Reproduce:**
1. Set up multiple maps with the same ID using the MapProvider.
2. Trigger the onMapMount function.

#### Error Message:
`Error: Multiple maps with the same id: ${id}
`
#### Relevant Line in Code:
[Line 25 in use-map.tsx](https://github.com/visgl/react-map-gl/blob/0ef118b045f59da38fa3153a0456787e8788073d/src/components/use-map.tsx#L25)

### **Question:**
I am assuming that this is the intended behavior but is there a recommended way implementation-wise to handle or avoid this error? I know that the docs mention that you can specify an ID for the map but I have not come across anything about the duplicate map issue. Any guidance or advice would be greatly appreciated.

**Comment by Pessimistress** [comment]
As the error suggests, you need to specify different ids if you are using multiple maps with the MapProvider.
